Account/FxAPushMessageHandler.swift

```diff
@@ -256,17 +256,13 @@ enum PushMessage: Equatable {
     }
 
     public static func ==(lhs: PushMessage, rhs: PushMessage) -> Bool {
-        guard lhs.messageType == rhs.messageType else {
-            return false
-        }
-
         switch (lhs, rhs) {
-        case (.deviceConnected(let lName), .deviceConnected(let rName)):
+        case let (.deviceConnected(lName), .deviceConnected(rName)):
             return lName == rName
-        case (.collectionChanged(let lList), .collectionChanged(let rList)):
+        case let (.collectionChanged(lList), .collectionChanged(rList)):
             return lList == rList
         default:
-            return true
+            return false
```

Comment:
This was a bug right?

Reply:
IIRC with this change enums with no associated values will not be equal. e.g. `.thisDeviceDisconnected != .thisDeviceDisconnected`.

If the `guard` in L259-L261 goes, and you add `case`s for each pair of messages in the enum, then defaulting to `false` would be OK.

Reply:
So are we comparing just the type or also the associated values?

Reply:
@jhugman ping to decide on this

Reply:
Pls dont look at this commit anymore. There's a branch that Farhan made for me. https://github.com/mozilla-mobile/firefox-ios/tree/adamschanges I think that you guys should look at the changes sooner rather than later cause there's a lot of them, I like to think generally for the better, and they are just gonna get stale.

Reply:
This change is actually wrong. Or like I just found it strange but kinda wish I left it the way it is as it seems to be too much of a distraction from the rest.